### PR TITLE
Feature for SageMaker: create_pipeline, list_pipelines

### PR DESCRIPTION
**Pull request by stiebels**
Added the following mocking functionality for `sagemaker`:
- `create_pipeline`
- `list_pipelines`

### CODE REVIEW

File: moto/sagemaker/models.py [hunk 1]
```diff
@@ -74,6 +74,49 @@ def response_object(self):
...
+        self.parallelism_configuration = parallelism_configuration
+        self.pipeline_definition_s3_location = pipeline_definition_s3_location
+
+        now_string = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        self.creation_time = now_string
+        self.last_modified_time = now_string
+        self.last_execution_time = now_string
+
+    @property
+    def response_object(self):
```
Comment (bblommers) [inline moto/sagemaker/models.py:109]: The `response_object` and `response_create` arent' used, as far as I can tell, so let's remove them

File: tests/test_sagemaker/test_sagemaker_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    assert isinstance(response, dict)
+    assert response["PipelineArn"].should.equal(
+        arn_formatter("pipeline", fake_pipeline_name, ACCOUNT_ID, TEST_REGION_NAME)
+    )
+
+
+def test_list_pipelines_none(sagemaker_client):
+    response = sagemaker_client.list_pipelines()
+    assert isinstance(response, dict)
+    assert response["PipelineSummaries"].should.be.empty
```
Comment (bblommers) [inline tests/test_sagemaker/test_sagemaker_pipeline.py:50]:
```suggestion
    response["PipelineSummaries"].should.be.empty
```
If you use the `should`, you don't need to assert - asserting whatever it `should` be is automatic.

There are quite a few assertions like that - I won't point them all out, but ideally we remove all of them.

File: tests/test_sagemaker/test_sagemaker_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+    created_before_timestamp = datetime.timestamp(created_before_datetime)
+    response = sagemaker_client.list_pipelines(CreatedBefore=created_before_timestamp)
+    assert response["PipelineSummaries"].should.be.empty
+
+
+def test_list_pipelines_invalid_values(sagemaker_client):
+    with pytest.raises(botocore.exceptions.ClientError):
+        _ = sagemaker_client.list_pipelines(MaxResults=200)  # Must be <= 100
+        _ = sagemaker_client.list_pipelines(NextToken="some-invalid-next-token")
```
Comment (bblommers) [inline tests/test_sagemaker/test_sagemaker_pipeline.py:174]: This doesn't work unfortunately. After the first command fails, we immediately jump out of the context, so the next three statements are ignored.

So this should either follow:
```
with pytest.raises():
    ....
with pytest.raises():
    ...
# etc
```
or we use `pytest.mark.parametrize` to execute the same test multiple times. I'll leave it up to you what works best!

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @stiebels, thanks for the PR! Looks good overall, just got a few minor nitpicks.

File: moto/sagemaker/models.py (hunk 1 above)
Comment (stiebels) [reply moto/sagemaker/models.py]: Thanks for the review and the comments! All change requests implemented.

File: tests/test_sagemaker/test_sagemaker_pipeline.py (hunk 2 above)
Comment (stiebels) [reply tests/test_sagemaker/test_sagemaker_pipeline.py:50]: flake8 only gives me a pass when having an `assert` before `.should.be.empty` statements. For other `should.have` (e.g. `should.have.length_of`), no assert seems to be required.

See here the workflow run on my forked repo: https://github.com/stiebels/moto/actions/runs/3709067440/jobs/6287306243

Therefore, I've left the `assert` in for `should.be.empty` and removed it where it doesn't seem to be needed.

**Review by bblommers — APPROVED** [review]
LGTM - Thanks again for adding these feature @stiebels!

**Comment by stiebels** [comment]
> LGTM - Thanks again for adding these feature @stiebels!

Thanks! Feel free to merge any time ("Only those with write access to this repository can merge pull requests.").

**Pull request merged**